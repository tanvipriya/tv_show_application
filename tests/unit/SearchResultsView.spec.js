import { mount } from "@vue/test-utils";
import SearchView from "@/views/SearchResultsView.vue"; // adjust path if needed

describe("SearchView.vue", () => {
    it("renders input with placeholder", () => {
        const wrapper = mount(SearchView);
        const input = wrapper.find("input");
        expect(input.exists()).toBe(true);
        expect(input.attributes("placeholder")).toBe("Search TV shows...");
    });

    it("emits update:modelValue when input value changes", async () => {
        const wrapper = mount(SearchView, {
            props: { modelValue: "" },
        });

        const input = wrapper.find("input");
        await input.setValue("Friends");

        // Assert emitted event
        const emitted = wrapper.emitted("update:modelValue");
        expect(emitted).toBeTruthy();
        expect(emitted[0]).toEqual(["Friends"]);
    });

    it("updates input value when modelValue prop changes", async () => {
        const wrapper = mount(SearchView, {
            props: { modelValue: "Breaking Bad" },
        });

        const input = wrapper.find("input");
        expect(input.element.value).toBe("Breaking Bad");

        // Update prop
        await wrapper.setProps({ modelValue: "Suits" });
        expect(input.element.value).toBe("Suits");
    });

    // âœ… Snapshot Test
    it("matches snapshot", () => {
        const wrapper = mount(SearchView, {
            props: { modelValue: "Stranger Things" },
        });
        expect(wrapper.html()).toMatchSnapshot();
    });
});
