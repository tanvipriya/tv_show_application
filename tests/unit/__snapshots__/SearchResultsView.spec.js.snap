// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchView.vue matches snapshot 1`] = `<div class="search-view"><input type="text" placeholder="Search TV shows..." class="search-input"></div>`;
